plugins/modules/route53.py

```diff
@@ -567,8 +627,8 @@ def main():
     if command_in == 'create' or command_in == 'delete':
         if alias_in and len(value_in) != 1:
             module.fail_json(msg="parameter 'value' must contain a single dns name for alias records")
-        if (weight_in is None and region_in is None and failover_in is None) and identifier_in is not None:
-            module.fail_json(msg="You have specified identifier which makes sense only if you specify one of: weight, region or failover.")
+        if not any([weight_in, region_in, failover_in, geo_location]) and identifier_in is not None:
```

Comment:
I guess there should be `all` instead of `any`.

Reply:
I tried `all` but with `all` it will return `True` only if all of the variables of `[weight_in, region_in, failover_in, geo_location]` are True/have value, it causes the condition to fail.
While with `any` it will return `True` when either of the vairable from `[weight_in, region_in, failover_in, geo_location]` are True/have value, causing condition to work as expected.

Can be tested by running this:
```
- name: route 53 test
  gather_facts: False
  hosts: localhost
  tasks:
    - name: Add new.routetest.com as an A record
      route53:
        state: present
        zone: routetest.com
        record: new.routetest.com
        type: A
        ttl: 7200
        value: 1.1.1.1
        identifier: routetest
        region: us-east-1
        geo_location:
          country_code: US
        wait: yes
      check_mode: false
```
With the above playbook, even though `region` and `geo_location` are mentioned, with `all` it returns
```
fatal: [localhost]: FAILED! => {"changed": false, "msg": "You have specified identifier which 
makes sense only if you specify one of: weight, region, geo_location or failover."}
```
While with `any`, it works as expected.